test/acceptance/tests.py

```diff
@@ -592,6 +623,210 @@ def test_file_upload(self):
         self.assertTrue(self.submission_page.has_file_uploaded)
 
 
+class FullWorkflowTest(OpenAssessmentTest):
+    """
+    Tests of complete workflows, combining multiple required steps together.
+    """
+    LEARNER_EMAIL = "learner@foo.com"
+    LEARNER_PASSWORD = "learner_password"
+
+    PEER_ASSESSMENT = [0, 0]
+    STAFF_AREA_PEER_ASSESSMENT = ['Poor', u'', u'0', u'5', u'Poor', u'', u'0', u'3']
+    PEER_ASSESSMENT_SCORE = 0
+    PEER_ASSESSMENT_STAFF_AREA_SCORE = "Final grade: 0 out of 8"
+
+    SELF_ASSESSMENT = [2, 3]
+    STAFF_AREA_SELF_ASSESSMENT = ['Good', u'5', u'5', u'Excellent', u'3', u'3']
+
+    SUBMITTED_ASSESSMENT = [0, 3]
+    STAFF_AREA_SUBMITTED = ['Poor', u'', u'0', u'5', u'Excellent', u'', u'3', u'3']
+
+    def setUp(self):
+        super(FullWorkflowTest, self).setUp('full_workflow', staff=True)
+        self.staff_area_page = StaffAreaPage(self.browser, self.problem_loc)
+
+    def login_user(self, learner, email=LEARNER_EMAIL, password=LEARNER_PASSWORD):
```

Comment:
I'd like to see these helper methods in a base class so that we can add further tests that perform submissions etc.